media/js/firefox/new/yandex/scene1.js

```diff
@@ -0,0 +1,231 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function() {
+    'use strict';
+
+    var Yandex = {
+        RUSSIA_COUNTRY_CODE: 'ru',
+        COOKIE_ID: 'firefox-yandex',
+        COOKIE_YANDEX_COHORT: 'firefox-yandex-cohort',
+        COOKIE_EXPIRATION_DAYS: 3,
+        YANDEX_SAMPLE_RATE: 0.9
+    };
+
+    var _client = Mozilla.Client;
+    var _geoTimeout;
+    var _requestComplete = false;
+
+    Yandex.getLocation = function() {
+        // should /country-code.json be slow to load,
+        // just show the regular messaging after 3 seconds waiting.
+        _geoTimeout = setTimeout(Yandex.onRequestComplete, 3000);
+
+        var xhr = new window.XMLHttpRequest();
+
+        xhr.onload = function(r) {
+            // make sure status is in the acceptable range
+            if (r.target.status >= 200 && r.target.status < 300) {
+                var country;
+
+                try {
+                    country = JSON.parse(r.target.responseText).country_code.toLowerCase();
+                } catch (e) {
+                    country = 'none';
+                }
+
+                Yandex.onRequestComplete(country);
+            }
+        };
+
+        xhr.open('GET', '/country-code.json');
+        // must come after open call above for IE 10 & 11
+        xhr.timeout = 2000;
+        xhr.send();
+    };
+
+    Yandex.hasGeoOverride = function(location) {
+        var loc = location || window.location.search;
+        if (loc.indexOf('geo=') !== -1) {
+            var urlRe = /geo=([a-z]{2})/i;
+            var match = urlRe.exec(loc);
+            if (match) {
+                return match[1].toLowerCase();
+            }
+            return false;
+        }
+        return false;
+    };
+
+    Yandex.verifyLocation = function(location) {
+        if (location) {
+            return location === Yandex.RUSSIA_COUNTRY_CODE;
+        }
+
+        return false;
+    };
+
+    Yandex.onRequestComplete = function(country) {
```

Comment:
If the getLocation XHR times out no value is passed to this function and so the cookie is set to `undefined`. Is that something we should be nervous about? Maybe it would be better to set it to `none` like on line 38?

Reply:
Also, it's great that the download button is still visible while the rest of this waits. 💯

Reply:
Hmm, good catch - I'll delve into this. Setting it to `none` sounds like the sane thing to do.